### PR TITLE
RomApp Easily Storing Matrices as Numpy Arrays in the RomManager

### PR DESCRIPTION
**Pull request by Rbravo555**
**📝 Description**
This PR allows the user to easily store the snapshots matrices of a given training campain by passing to the Fit() method of the RomManager the following parameters:
- store_all_snapshots=False/True
- store_fom_snapshots=False/True
- store_rom_snapshots=False/True
- store_hrom_snapshots=False/True
- store_residuals_projected=False/True
The matrices are stored in a folder named "./SnapshotsMatrices/" in the current directory

**🆕 Changelog**
- Added the parameters with default value to the RomManager.Fit() method 
- Added the method RomManager._StoreSnapshotsMatrix(self,string_numpy_array_name, numpy_array)

### CODE REVIEW

**Review by SADPR — COMMENTED** [review]
Looks good to me.

**Review by NicolasSR — COMMENTED** [review]
Seems correct to me aswell

File: applications/RomApplication/python_scripts/rom_manager.py [hunk 1]
```diff
@@ -517,5 +535,15 @@ def _GetDefaulHromTrainingParameters(self):
 
 
 
+    def _StoreSnapshotsMatrix(self, string_numpy_array_name, numpy_array):
+
+        # Define the directory and file path
+        directory = './SnapshotsMatrices'
```
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py:541]: Maybe we should do this user definable?

File: applications/RomApplication/python_scripts/rom_manager.py [hunk 2]
```diff
@@ -517,5 +535,15 @@ def _GetDefaulHromTrainingParameters(self):
...
+        file_path = os.path.join(directory, string_numpy_array_name + '.npy')
```
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/rom_manager.py]:
```suggestion
        file_path = os.path.join(directory, f'{string_numpy_array_name}.npy')
```

**Review by rubenzorrilla — COMMENTED** [review]
Two minor details, aside of these changes look good to me.

File: applications/RomApplication/python_scripts/rom_manager.py (hunk 1 above)
Comment (Rbravo555) [reply applications/RomApplication/python_scripts/rom_manager.py:541]: We shall be careful not to hyper-parametrise the RomManager. That is one of the reasons I have opted not to pass an extra parameter to the RomManager itself, but pass an optional parameter to its Fit() method. 

Having said that, in an upcoming PR which is being prepared by @SADPR(#10990), multiple ProjectParameters.json will be usable, and therefore the path names will have to be provided. If we can leave it hard-coded (although quite evocative) for now, we will address the issue of allowing users to define their own folder names in that PR.

Comment (rubenzorrilla) [reply applications/RomApplication/python_scripts/rom_manager.py:541]: Fair enough.

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**